packages/headless/src/controllers/triggers/headless-redirection-trigger.ts

```diff
@@ -0,0 +1,115 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {Engine} from '../../app/headless-engine';
+import {
+  ConfigurationSection,
+  RedirectionSection,
+} from '../../state/state-sections';
+import {configuration, redirection} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {
+  validateInitialState,
+  validateOptions,
+} from '../../utils/validate-payload';
+
+export interface RedirectionInitialState {
+  /**
+   * The initial url used for the redirection.
+   **/
+  redirectTo?: string;
+}
+
+export interface RedirectionOptions {
+  /**
+   * The function used to handle whenever the `Redirection` controller's state's `redirectTo` value changes.
+   */
+  function: Function;
+}
+
+export interface RedirectionProps {
+  /**
+   * The options for the `Redirection` controller.
+   */
+  options?: RedirectionOptions;
+  /**
+   * The initial state that should be applied to the `Redirection` controller.
+   */
+  initialState?: RedirectionInitialState;
+}
+
+const optionsSchema = new Schema({
+  function: new Function(),
```

Comment:
I have a compilation error on this line that I don't understand: `Type 'Function' is missing the following properties from type 'SchemaValue<unknown>': validate, default, required`

Reply:
We don't have anything in Bueno I think, it only needs to be defined, perhaps this would work, it would have to be tested
```suggestion
  onRedirect: new Value({required: true}),
```